app/src/lib/stats/stats-store.ts

```diff
@@ -1772,6 +1795,55 @@ export class StatsStore implements IStatsStore {
     }))
   }
 
+  // Generates the stat field name for the given PR review type and suffix.
+  private getStatFieldForRequestReviewState(
+    reviewType: ValidNotificationPullRequestReviewState,
+    suffix: PullRequestReviewStatFieldSuffix
+  ): PullRequestReviewStatField {
+    const infixMap: Record<
+      ValidNotificationPullRequestReviewState,
+      PullRequestReviewStatFieldInfix
+    > = {
+      CHANGES_REQUESTED: 'ChangesRequested',
+      APPROVED: 'Approved',
+      COMMENTED: 'Commented',
+    }
+
+    return `pullRequestReview${infixMap[reviewType]}${suffix}`
+  }
+
+  // Generic method to record stats related to Pull Request review notifications.
+  private recordPullRequestReviewStat(
+    reviewType: ValidNotificationPullRequestReviewState,
+    suffix: PullRequestReviewStatFieldSuffix
+  ) {
+    const statField = this.getStatFieldForRequestReviewState(reviewType, suffix)
+    return this.updateDailyMeasures(
+      m => ({ [statField]: m[statField] + 1 } as any)
+    )
```

Comment:
This cast here is a bummer but I'm not sure what the best way to get rid of it would be. Copying and assigning manually works but isn't as succinct

```ts
return this.updateDailyMeasures(m => {
  const copy = { ...m }
  copy[statField] = copy[statField] + 1
  return copy
})
```

Reply:
Yep I think I'm gonna choose succinctness here, also because I'd like to avoid that extra copy. Thanks for looking into this and for your feedback! 🙇‍♂️  Glad to know I'm learning useful stuff from those type challenges 😂